### ISSUE TITLE
Solver issue (singularities)

### ISSUE THREAD

**Issue by jauhl** [goessner/mec2#13]
There seems to be an issue with the dependend constraint d from the following mechanism
```json
{
  "id":"linkage",
  "nodes": [
    { "id":"A","x":0,"y":0,"base":true },
    { "id":"B","x":300,"y":0,"base":true },
    { "id":"C","x":0,"y":75 },
    { "id":"D","x":40,"y":200 },
    { "id":"E","x":185,"y":245 }
  ],
  "constraints": [
    { "id":"a","p1":"A","p2":"C","len":{ "type":"const" } },
    { "id":"b","p1":"C","p2":"D","len":{ "type":"const" } },
    { "id":"c","p1":"B","p2":"D","len":{ "type":"const" } },
    { "id":"d","p1":"C","p2":"E","len":{ "type":"const" },"ori":{ "type":"ref","ref":"b" } }
  ]
}
```
Try dragging the linkage in a position where a and b coincide. You can't.
The issue is gone when you increase the angle between b and d or simply remove d from the model.

**Comment by goessner** [comment]
yes ... confirmed.

Should be `ref`-related. Must wait until reimplementation of references.

**Comment by jauhl** [comment]
Another case which i blame on references is this one:
```json
{
  "id":"linkage",
  "nodes": [
    { "id":"A","x":212,"y":238,"base":true },
    { "id":"B","x":324,"y":119,"base":true },
    { "id":"C","x":493,"y":481 },
    { "id":"D","x":557,"y":445 },
    { "id":"E","x":709,"y":363 }
  ],
  "constraints": [
    { "id":"b","p1":"B","p2":"D","len":{ "type":"const" } },
    { "id":"c","p1":"C","p2":"D","len":{ "type":"const" } },
    { "id":"a","p1":"A","p2":"C","len":{ "type":"const" },"ori":{ "type":"drive","Dt":1,"Dw":6.283185307179586,"input":true } },
    { "id":"d","p1":"D","p2":"E","len":{ "type":"const" },"ori":{ "type":"ref","ref":"c" } }
  ]
}
```

Load it in [mecEdit](https://jauhl.github.io/mecEdit/mecEdit.html). Using the range-input will destroy the model so that even `model.reset()` doesn't help anymore. It needs to be completely reinitialized  (e.g. keyboard <kbd>e</kbd> -> "Apply changes").
But when you delete constraint d(c) and then use the drive-slider, the model becomes invalid but stays resetable as intended.

**Comment by jauhl** [comment]
resolved with c0f3a1744798566af61dd4e52be62c053661c896
